rust/src/transformations/make_stable_expr/expr_discrete_quantile_score/mod.rs

```diff
@@ -0,0 +1,208 @@
+            )
+        }
+        dtype => {
+            return fallible!(
+                MakeTransformation,
+                "Expected numeric data type, found {:?}",
+                dtype
+            );
+        }
+    }?;
+
+    let margin = middle_domain.active_margin()?.clone();
+
+    let mpl = margin
+        .max_partition_length
+        .ok_or_else(|| err!(MakeTransformation, "Must know max_partition_length"))?;
+
+    let constants = score_candidates_constants::<u64>(Some(mpl as u64), alpha)?;
+    let (alpha_num, alpha_den, _) = constants.clone();
+
+    t_prior
+        >> Transformation::<_, _, PartitionDistance<MI>, Parallel<LInfDistance<_>>>::new(
+            middle_domain.clone(),
+            middle_domain,
+            Function::new_expr(move |input_expr| {
+                apply_plugin(
+                    input_expr,
+                    expr.clone(),
+                    DQScoreArgs {
+                        alpha,
+                        candidates: candidates.clone(),
+                        constants: Some(constants),
+                    },
+                )
+            }),
+            middle_metric,
+            Parallel(LInfDistance::new(false)),
+            StabilityMap::new_fallible(move |(l0, _, li)| {
+                let out_li = f64::inf_cast(score_candidates_map(
+                    alpha_num,
+                    alpha_den,
+                    margin.public_info == Some(MarginPub::Lengths),
+                )(li)?)?;
+                Ok((*l0, out_li))
+            }),
+        )?
+}
+
+/// Determine if the given expression is a discrete quantile score expression.
+///
+/// # Arguments
+/// * `expr` - The expression to check
+///
+/// # Returns
+/// If matched, ipnut expression and discrete quantile score arguments
+pub(crate) fn match_dq_score(expr: &Expr) -> Fallible<Option<(&Expr, DQScoreArgs)>> {
+    let Some((score_input, args)) = match_plugin(expr, "dq_score")? else {
+        return Ok(None);
+    };
+
+    let [score_input] = <&[_; 1]>::try_from(score_input.as_slice()).map_err(|_| {
+        err!(
+            MakeTransformation,
+            "dq_score expects a single input expression"
+        )
+    })?;
+
+    Ok(Some((score_input, args)))
+}
+
+#[cfg(test)]
+pub mod test_expr_quantile {
+    use super::*;
+    use polars::prelude::*;
+
+    use crate::{
+        core::PrivacyNamespaceHelper,
+        domains::{AtomDomain, LazyFrameDomain, Margin, SeriesDomain},
+        metrics::SymmetricDistance,
+    };
+
+    pub fn get_quantile_test_data() -> Fallible<(LazyFrameDomain, LazyFrame)> {
+        let pub_key_margin = Margin::new()
+            .with_max_partition_length(1000)
+            .with_public_keys();
+
+        let lf_domain = LazyFrameDomain::new(vec![
+            SeriesDomain::new("A", AtomDomain::<i32>::default()),
+            SeriesDomain::new("B", AtomDomain::<f64>::default()),
```

Comment:
Slight preference for naming columns to reflect content, if it matters for the behavior.

Reply:
forgot about this, I'll make it use the new test data function.